### PR TITLE
httpy-cli: init at 1.1.0

### PR DESCRIPTION
**Pull request by hatch01**
## Description of changes

add the package httpy-cli

## Things done

Create de derivation for httpy-cli.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by BCNelson — CHANGES_REQUESTED** [review]
Is there a reason that you are doing version 1.0.0 rather than the version 1.1.0? Think that we should have the most resent version

File: pkgs/top-level/all-packages.nix
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:32176]:
```suggestion
```

File: pkgs/applications/networking/httpy-cli/default.nix
Comment (AndersonTorres) [inline pkgs/applications/networking/httpy-cli/default.nix:1]: Move this file to `pkgs/by-name/ht/httpy-cli/package.nix`.

File: pkgs/applications/networking/httpy-cli/default.nix
Comment (AndersonTorres) [inline pkgs/applications/networking/httpy-cli/default.nix:24]:
```suggestion
    description = "Modern, user-friendly, programmable command-line HTTP client for the API";
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by hatch01** [comment]
> Is there a reason that you are doing version 1.0.0 rather than the version 1.1.0? Think that we should have the most resent version

It was a mistake because of releases on GitHub, but I switched to 1.1.0.

**Comment by BCNelson** [comment]
Okay I think you will need to update the git commit message and pr name. Then you are good to go.

**Comment by hatch01** [comment]
> Okay I think you will need to update the git commit message and pr name. Then you are good to go.

It's done

**Review by BCNelson — APPROVED** [review]

**Comment by hatch01** [comment]
@AndersonTorres does it look good now ?

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [inline pkgs/by-name/ht/httpy-cli/package.nix:23]: Missing `meta.mainProgram`.

**Review by drupol — CHANGES_REQUESTED** [review]

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix:23]: it's done !

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix:23]: its done

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [inline pkgs/by-name/ht/httpy-cli/package.nix:12]: Don't reuse `pname`, use a string literal instead.

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix]: Is this what you asked for?
Because I just copied the examples from the documentation here https://nixos.wiki/wiki/Packaging/Python#install_binary_release

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [inline pkgs/by-name/ht/httpy-cli/package.nix:13]: You can reuse the `version` here.

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix]: okay, but why not inheriting it?

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [reply pkgs/by-name/ht/httpy-cli/package.nix]: We usually do not reuse the derivation attribute `pname`, however, we reuse the `version`.

Comment (drupol) [reply pkgs/by-name/ht/httpy-cli/package.nix]: `repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [inline pkgs/by-name/ht/httpy-cli/package.nix:13]: `inherit version;` would have been fine too, but OK I won't complain.

**Review by drupol — APPROVED** [review]

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [inline pkgs/by-name/ht/httpy-cli/package.nix:29]: Oh I just notice that the maintainer is not you, is it intended?

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix:29]: it's me Eymeric (github account hatch01)

**Comment by hatch01** [comment]
What do I need to wait for merge?

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [reply pkgs/by-name/ht/httpy-cli/package.nix:29]: Oh... sorry about that. The confusing part is that your Nix contributor pseudo is different from your Github account. All good anyway!

**Comment by drupol** [comment]
Let's wait for the CI tests to be completely finished and I'll merge it.

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/ht/httpy-cli/package.nix:9]: We cannot use pyproject = true here?

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (SuperSandro2000) [inline pkgs/by-name/ht/httpy-cli/package.nix:23]: Please add a pythonImportsCheck

Can we execute the tests https://github.com/knid/httpy/tree/main/tests ?

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix:9]:
```
this derivation will be built:
  /nix/store/bg3qa1a09sig1sw0gmvr496jwfnxgw80-python3.11-httpy-cli-1.1.0.drv
these 5 paths will be fetched (0.14 MiB download, 0.84 MiB unpacked):
  /nix/store/hb6bxbfjdzysgfv9r5r6w8zyvpfynans-pypa-build-hook.sh
  /nix/store/m3smb9y6q0hvp1z3gwzh4mzz56swrb45-python-runtime-deps-check-hook.sh
  /nix/store/03gjlnbiwd01mxa42y5r7dchvnhy61cb-python3.11-build-1.0.3
  /nix/store/5gfnjgbnysagg726a7s8jf52zrpwn1dw-python3.11-packaging-23.2
  /nix/store/pnaj399xsswrxg938lgx8f4xi1b8ryfl-python3.11-pyproject-hooks-1.0.0
copying path '/nix/store/5gfnjgbnysagg726a7s8jf52zrpwn1dw-python3.11-packaging-23.2' from 'https://cache.nixos.org'...
copying path '/nix/store/pnaj399xsswrxg938lgx8f4xi1b8ryfl-python3.11-pyproject-hooks-1.0.0' from 'https://cache.nixos.org'...
copying path '/nix/store/m3smb9y6q0hvp1z3gwzh4mzz56swrb45-python-runtime-deps-check-hook.sh' from 'https://cache.nixos.org'...
copying path '/nix/store/03gjlnbiwd01mxa42y5r7dchvnhy61cb-python3.11-build-1.0.3' from 'https://cache.nixos.org'...
copying path '/nix/store/hb6bxbfjdzysgfv9r5r6w8zyvpfynans-pypa-build-hook.sh' from 'https://cache.nixos.org'...
building '/nix/store/bg3qa1a09sig1sw0gmvr496jwfnxgw80-python3.11-httpy-cli-1.1.0.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pypa-build-hook
Using pypaBuildPhase
Sourcing python-runtime-deps-check-hook
Using pythonRuntimeDepsCheckHook
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Running phase: unpackPhase
unpacking source archive /nix/store/8yyj0dkn1p1laqb04vnn0nlkh0487ax5-httpy-cli-1.1.0.tar.gz
source root is httpy-cli-1.1.0
setting SOURCE_DATE_EPOCH to timestamp 1680789829 of file httpy-cli-1.1.0/setup.cfg
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing pypaBuildPhase
Creating a wheel...
* Getting build dependencies for wheel...

Traceback (most recent call last):
  File "/nix/store/pnaj399xsswrxg938lgx8f4xi1b8ryfl-python3.11-pyproject-hooks-1.0.0/lib/python3.11/site-packages/pyproject_hooks/_impl.py", line 321, in _call_hook
    raise BackendUnavailable(data.get('traceback', ''))
pyproject_hooks._impl.BackendUnavailable: Traceback (most recent call last):
  File "/nix/store/pnaj399xsswrxg938lgx8f4xi1b8ryfl-python3.11-pyproject-hooks-1.0.0/lib/python3.11/site-packages/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
    obj = import_module(mod_path)
          ^^^^^^^^^^^^^^^^^^^^^^^
  File "/nix/store/sxr2igfkwhxbagri49b8krmcqz168sim-python3-3.11.8/lib/python3.11/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
ModuleNotFoundError: No module named 'setuptools'

ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
error: builder for '/nix/store/bg3qa1a09sig1sw0gmvr496jwfnxgw80-python3.11-httpy-cli-1.1.0.drv' failed with exit code 1;
       last 10 log lines:
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
       >   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
       >   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
       >   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
       >   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
       > ModuleNotFoundError: No module named 'setuptools'
       >
       > ERROR Backend 'setuptools.build_meta:__legacy__' is not available.
       For full logs, run 'nix log /nix/store/bg3qa1a09sig1sw0gmvr496jwfnxgw80-python3.11-httpy-cli-1.1.0.drv'.

```

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix]: My bad I found the fix myself.

File: pkgs/by-name/ht/httpy-cli/package.nix
Comment (drupol) [reply pkgs/by-name/ht/httpy-cli/package.nix:24]: @hatch01 Have you addressed this comment?

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix:24]: Oh my bad I forgot to push

Comment (hatch01) [reply pkgs/by-name/ht/httpy-cli/package.nix:24]: this was a mistake I added it !

**Comment by hatch01** [comment]
It seems complete to me now.

**Comment by drupol** [comment]
> It seems complete to me now.

mmmh when I look at other python derivation, `pythonImportsCheck` only contains the module it is importing. Therefore, I would ask to only have `httpy` in `pythonImportsCheck`.

**Comment by hatch01** [comment]
Yeah, it seems better I have done it.

**Review by drupol — APPROVED** [review]
LGTM now, thanks !

Waiting for the CI to be done before merging now.

**Pull request merged**
